improver/utilities/spatial.py

```diff
@@ -490,6 +487,52 @@ def maximum_within_vicinity(self, cube: Cube) -> Cube:
             max_cube.data = max_data
         return max_cube
 
+    def set_grid_point_radius(self, cube):
+        """
+        Set the grid_point_radius if it has not been provided in this form.
+        If a radius has been provided as a physical distance, convert this into
+        a number of grid points. If the radius distance is 0, set the grid point
+        radius to 0.
+
+        Args:
+            cube:
+                Thresholded cube.
+        """
+        if self.grid_point_radius is None:
+            self.native_grid_point_radius = False
+            if self.radius != 0:
+                return distance_to_number_of_grid_cells(cube, self.radius)
+            return 0
+        else:
+            self.native_grid_point_radius = True
+            return self.grid_point_radius
+
+    def _add_vicinity_coordinate(self, cube):
+        """
+        Add a coordinate to the returned cube that records the vicinity radius
+        that has been applied to the data.
+
+        Args:
+            cube:
+                Vicinity processed cube.
+        """
+        if self.native_grid_point_radius:
```

Comment:
Do we actually need to support native grid point radius now?

Reply:
Yes, this was added for precip phase change to provide a non-zero neighbourhood size when calculating maximum orography in neighbourhood. This is still being used, e.g. [here](https://github.com/MetOffice/improver_suite/blob/2382e71199b10a8bb5987e74e62d7b86060e31f1/lib/python/mo/settings/app/phase_change_level.py#L25). We've added another step after that to fix up any remaining NaNs, but the neighbourhood approach is preferable to the applying only the latter.

Reply:
Ok fair enough